### PR TITLE
Add model-mgmt UT

### PR DESCRIPTION
**Pull request by novaturient95**

### CODE REVIEW

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: training-model-mgmt-unittests
+
+on:
+  # push:
+  #   branches:
+  #     - release
+  pull_request:
+    branches:
+      - main
+      - ayushmishra/update_mlflow_convertor_params
```
Comment (skasturi) [inline .github/workflows/training-model-mgmt-unittests.yaml:10]: Remove this before merging?

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    paths:
+      - assets/training/model_management/*
```
Comment (skasturi) [inline .github/workflows/training-model-mgmt-unittests.yaml:12]: Include common/deploy or registration as well?

**Review by skasturi — APPROVED** [review]
Please address comments.

**Review by vizhur — DISMISSED** [review]

File: assets/training/model_management/tests/dev_requirements.txt [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# src
+azure-ai-ml
+azure-identity
+scipy==1.5.3
+numpy==1.24.4
+azureml-evaluate-mlflow
```
Comment (skasturi) [inline assets/training/model_management/tests/dev_requirements.txt:6]: Do we need to add the macro corresponding to latest version?

**Review by skasturi — APPROVED** [review]

File: .github/workflows/training-model-mgmt-unittests.yaml (hunk 2 above)
Comment (elesel) [inline .github/workflows/training-model-mgmt-unittests.yaml:12]:
```suggestion
      - assets/training/model_management/**
```

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        uses: actions/checkout@v3
+
+      - name: Use Python 3.8
+        uses: actions/setup-python@v4
+        with:
+          python-version: '==3.8'
+
+      - name: Install dev requirements
+        run: |
+          apt-get update && apt-get upgrade && apt-get install -y build-essential
```
Comment (elesel) [inline .github/workflows/training-model-mgmt-unittests.yaml:44]: Are these packages required?

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+          pip install -r $model_mgmt_test_req
+
+      - name: Copy dir
+        run: |
+          cp -r $model_mgmt_src_dir $model_mgmt_unittests_dir
+          ls -l $model_mgmt_unittests_dir
+
+      - name: Execute tests
+        run: python -m pytest $model_mgmt_unittests_dir --tb=native --junitxml=./TEST-TEST.xml -x -n 1 -ra --show-capture=no
```
Comment (elesel) [inline .github/workflows/training-model-mgmt-unittests.yaml:53]: Do you want to publish test results to the PR? See [this](https://github.com/Azure/azureml-assets/blob/e4ddb9853ab1995cf0ea97df2c094322174b6190/.github/workflows/scripts-test.yaml#L79-L84) for an example.

File: .github/workflows/training-model-mgmt-unittests.yaml (hunk 4 above)
Comment (novaturient95) [reply .github/workflows/training-model-mgmt-unittests.yaml:44]: yes @elesel

File: .github/workflows/training-model-mgmt-unittests.yaml (hunk 5 above)
Comment (novaturient95) [reply .github/workflows/training-model-mgmt-unittests.yaml:53]: Thanks @elesel. Added

Comment (novaturient95) [reply .github/workflows/training-model-mgmt-unittests.yaml]: workflow is failing with: 

image

Are you aware why this error is coming?

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  model_mgmt_unittests_dir: assets/training/model_management/tests/unittests
+  model_mgmt_test_req: assets/training/model_management/tests/dev_requirements.txt
+  pytest_reports: pytest-reports
+
+jobs:
+  run_unit_tests:
+    name: Run
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (elesel) [inline .github/workflows/training-model-mgmt-unittests.yaml:30]:
```suggestion
permissions:
    # Required for EnricoMi/publish-unit-test-result-action
    checks: write
    issues: read
    pull-requests: write

    steps:
```

File: .github/workflows/training-model-mgmt-unittests.yaml (hunk 2 above)
Comment (novaturient95) [reply .github/workflows/training-model-mgmt-unittests.yaml]: updated

Comment (novaturient95) [reply .github/workflows/training-model-mgmt-unittests.yaml]: For register and model-local-validation

File: .github/workflows/training-model-mgmt-unittests.yaml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      - assets/common/src/model_registration/**
+      - assets/common/components/mlflow_model_local_validation/**
+      - assets/common/src/mlflow_model_local_validation/**
+      - .github/workflows/training-model-mgmt-unittests.yaml
+  workflow_dispatch:
+    inputs:
+      asset_dirs:
+        description: Asset directories
+        default: assets
+        required: true
```
Comment (elesel) [inline .github/workflows/training-model-mgmt-unittests.yaml:19]: It doesn't look like you need the `asset_dirs` input:
```suggestion
  workflow_dispatch:
```

**Review by vizhur — DISMISSED** [review]

**Review by skasturi — APPROVED** [review]

**Review by vikasagrawal-ms — APPROVED** [review]

**Review by elesel — APPROVED** [review]

**Pull request merged**